src/main/java/net/sf/jabref/logic/importer/fetcher/IsbnFetcher.java

```diff
@@ -3,68 +3,53 @@
 import java.net.MalformedURLException;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.util.Optional;
 
-import net.sf.jabref.logic.formatter.bibtexfields.ClearFormatter;
-import net.sf.jabref.logic.formatter.bibtexfields.NormalizePagesFormatter;
-import net.sf.jabref.logic.help.HelpFile;
 import net.sf.jabref.logic.importer.FetcherException;
-import net.sf.jabref.logic.importer.IdBasedParserFetcher;
 import net.sf.jabref.logic.importer.ImportFormatPreferences;
-import net.sf.jabref.logic.importer.Parser;
-import net.sf.jabref.logic.importer.fileformat.BibtexParser;
-import net.sf.jabref.logic.l10n.Localization;
-import net.sf.jabref.logic.util.ISBN;
-import net.sf.jabref.model.cleanup.FieldFormatterCleanup;
 import net.sf.jabref.model.entry.BibEntry;
-import net.sf.jabref.model.entry.FieldName;
 
-import org.apache.http.client.utils.URIBuilder;
+import org.jsoup.helper.StringUtil;
 
 /**
- * Fetcher for ISBN using http://www.ebook.de.
+ * Fetcher for ISBN trying ebook.de first and then chimbori.com
  */
-public class IsbnFetcher implements IdBasedParserFetcher {
-
-    private ImportFormatPreferences importFormatPreferences;
+public class IsbnFetcher extends AbstractIsbnFetcher {
 
     public IsbnFetcher(ImportFormatPreferences importFormatPreferences){
-        this.importFormatPreferences = importFormatPreferences;
-    }
-
-    @Override
-    public String getName() {
-        return "ISBN";
+        super(importFormatPreferences);
     }
 
+    /**
+     * Method never used
+     */
     @Override
-    public HelpFile getHelpPage() {
-        return HelpFile.FETCHER_ISBN_TO_BIBTEX;
+    public URL getURLForID(String identifier) throws URISyntaxException, MalformedURLException, FetcherException {
+        return null;
     }
 
     @Override
-    public URL getURLForID(String identifier) throws URISyntaxException, MalformedURLException, FetcherException {
-        ISBN isbn = new ISBN(identifier);
-        if (!isbn.isValid()) {
-            throw new FetcherException(Localization.lang("Invalid_ISBN:_'%0'.", identifier));
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        if (StringUtil.isBlank(identifier)) {
+            return Optional.empty();
         }
 
-        URIBuilder uriBuilder = new URIBuilder("http://www.ebook.de/de/tools/isbn2bibtex");
-        uriBuilder.addParameter("isbn", identifier);
-        return uriBuilder.build().toURL();
-    }
+        this.ensureThatIsbnIsValid(identifier);
 
-    @Override
-    public Parser getParser() {
-        return new BibtexParser(importFormatPreferences);
+        IsbnViaEbookDeFetcher isbnViaEbookDeFetcher = new IsbnViaEbookDeFetcher(importFormatPreferences);
```

Comment:
I don't like the architecture that a fetcher controls other fetcher. All of them should be completely independent and their working-together should be controlled at a higher level. Moreover, I would allow the user to choose between "ISBN (ebook.de)" and "ISBN (Chimbori/Amazon)", or is the quality of ebook's metadata really superior?